addons/purchase/tests/test_purchase_invoice.py

```diff
@@ -328,6 +328,49 @@ def test_vendor_bill_analytic_account_default_change(self):
         aml = self.env['account.move.line'].search([('purchase_line_id', '=', purchase_order.order_line.id)])
         self.assertRecordValues(aml, [{'analytic_account_id': analytic_account_manual.id}])
 
+    def test_vendor_bill_analytic_account_product_change(self):
```

Comment:
Does this test do something else than the one added here?
https://github.com/odoo/odoo/commit/171a6b93a26af3fd52e52c95916a11c1b7e79ad7

Reply:
Yes, except for the last assert.

For the previous assert's, it tests whether we change the product on a move_line, the default analytic account is also changed automatically. 

The tests does not pass without the fix, the tests pass with the fix.